### DISCUSSION TITLE
Kedro + Ibis + Local BigQuery emulation

### DISCUSSION THREAD

**Discussion by lvijnck** [ibis-project/ibis#9258] [Q&A] [answered]
Hi there,

Thanks for the great tooling! 

For my test setup, I'm looking for a way to setup Ibis to leverage the [BigQuery emulator](https://github.com/goccy/bigquery-emulator). I've used the TableDatset from [this](https://kedro.org/blog/building-scalable-data-pipelines-with-kedro-and-ibis) blogpost.

From what I can tell, there seems to be no direct way to customize `api_endpoint `.

Any ideas?

Thanks in advance!

**Comment by gforsyth** [comment, marked as answer]
Hi @lvijnck -- there may be waaay more involved to get this working, but you can pass an existing `bigquery.Client` object to `ibis.bigquery.connect(client=...)` -- and Ibis will use the existing client (if it exists) instead of creating one.
